File: MotionLayout/app/src/main/java/com/example/taling11/MainActivity.java
package com.example.taling11;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btnGOMotionActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnGOMotionActivity = findViewById(R.id.btn_goMotionActivity);
        btnGOMotionActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,MotionActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: MVC/app/src/main/java/com/puresoftware/mvc/MainActivity.kt
package com.puresoftware.mvc

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast


// MVC중 Controller. View와 Model을 이어줄 중심부
class MainActivity : AppCompatActivity() {

    private lateinit var user: User // Model
    lateinit var etUserName: EditText
    lateinit var etPassword: EditText
    lateinit var loginBtn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)// 확실한 개념을 위해 ViewBinding을 하지 않음.

        user = User() // Model Class 초기화

        etUserName = findViewById(R.id.etUserName)
        etPassword = findViewById(R.id.etPassword)
        loginBtn = findViewById(R.id.loginBtn)

        // 프레젠테이션 로직
        loginBtn.setOnClickListener {
            val isLoginSuccessful = user.login(
                etUserName.text.toString(),
                etPassword.text.toString(),
            )

            if (isLoginSuccessful) {
                Toast.makeText(this, "${user.userName} Login Successful", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "Login Failed", Toast.LENGTH_SHORT).show()
            }
        }
    }
}
// 원리
// 구조는 Model, Controller, View
// Model Class는 여러 정보가 담긴 객체
// View는 layout.xml
// Controller는 Activity 또는 Fragment 또는 Adapter등
// 결론은 Activity에서 View의 Method로 Model을 연결
